src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/alerts/AlertMappingService.kt

```diff
@@ -35,12 +38,49 @@ class AlertMappingService(val repository: AlertsMappingRepository) {
   }
 
   @Transactional
-  suspend fun createMappings(mappings: List<AlertMappingDto>) =
+  suspend fun createMappings(mappings: List<AlertMappingDto>) {
     repository.saveAll(mappings.map { it.fromDto() }).collect()
+    // respect backward compatibility, offenderNo can be null still until we move to method below
+    mappings.firstOrNull { it.offenderNo != null }?.apply {
+      prisonerMappingRepository.save(
+        AlertPrisonerMapping(
+          offenderNo = this.offenderNo!!,
+          count = mappings.size,
+          mappingType = this.mappingType,
+          label = this.label,
+        ),
+      )
+    }
+  }
+
+  @Transactional
```

Comment:
we need this new version so we can mark a prisoner as migrated even if they have no alerts